packages/core/src/config.ts

```diff
@@ -1480,14 +1482,14 @@ async function composeLibRsbuildConfig(
     pkgJson,
   );
   const userExternalsConfig = composeExternalsConfig(
-    format!,
+    format,
     config.output?.externals,
   );
   const {
     config: autoExtensionConfig,
     jsExtension,
     dtsExtension,
-  } = composeAutoExtensionConfig(config, autoExtension, pkgJson);
+  } = composeAutoExtensionConfig(config, format, autoExtension, pkgJson);
```

Comment:
`config` variable should be renamed to `rawLibConfig` so the other `format` variables are implied to be **resolved**, we could change it in the future.